source/extensions/filters/network/common/redis/client_impl.cc

```diff
@@ -94,9 +98,15 @@ PoolRequest* ClientImpl::makeRequest(const RespValue& request, PoolCallbacks& ca
 
   const bool empty_buffer = encoder_buffer_.length() == 0;
 
-  pending_requests_.emplace_back(*this, callbacks);
+  std::string to_lower_command(redis_command_stats_->getCommandFromRequest(request));
```

Comment:
hmmm....should we do the lower-casing inside getCommandFromRequest? Or is it useful in some contexts to have non-lower-cased names?

I *think* from reading this PR that you don't use the command name at all unless stats are enabled. Is that right? In that case I think we should skip the lower-casing and string-copying if enableCommandsStats() is false.

Reply:
Ok, in my update to the PR, I've added an "unused" statname, and used that for this ie

```
  Stats::StatName stat_name;
  if (config_.enableCommandStats()) {
    // Only lowercase command and get stat_name if we enable command stats
    std::string to_lower_command(redis_command_stats_->getCommandFromRequest(request));
    to_lower_table_.toLowerCase(to_lower_command);
    stat_name = redis_command_stats_->getStatName(to_lower_command);

    redis_command_stats_->updateStatsTotal(stat_name);
  } else {
    // If disabled, we use a placeholder stat name "unused" that is not used
    stat_name = redis_command_stats_->getUnusedStatName();
  }
```